Website/app_code/code/Blog.cs

```diff
@@ -63,10 +63,10 @@ public static Post CurrentPost
         {
             if (HttpContext.Current.Items["currentpost"] == null && !string.IsNullOrEmpty(CurrentSlug))
             {
-                var post = Storage.GetAllPosts().FirstOrDefault(p => p.Slug == CurrentSlug);
+                var post = GetVisiblePosts().FirstOrDefault(p => p.Slug == CurrentSlug);
 
-                if (post != null && (post.IsPublished || HttpContext.Current.User.Identity.IsAuthenticated || (HttpContext.Current.Request.QueryString["key"] ?? string.Empty).Equals(post.ID, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
Ah! That's a very useful feature indeed! Thanks a lot Sam! I just opened a PR to re-enable it.